### PR TITLE
Youtube linking doesn't work  fixed

### PR DESCRIPTION
**Pull request by UmairMukhtar**
Issue Fixed 
https://github.com/eLearningDAO/POCRE/issues/292

### CODE REVIEW

**Comment by UmairMukhtar** [comment]
[Video_2023_03_01-4.webm](https://user-images.githubusercontent.com/34018917/222223715-0a10ec50-c508-497e-8b6d-cd2bfac2a5b4.webm)

**Review by AltiMario — CHANGES_REQUESTED** [review]
It doesn't work for me
image

File: app/api/src/utils/getSupportedFileTypeFromLink.ts [hunk 1]
```diff
@@ -12,11 +13,30 @@
...
+  return (urlArray[2] !== undefined) ? urlArray[2].split(/[^0-9a-z_\-]/i)[0] : urlArray[0];
+}
+
+
+export const isValidYoutubeID = async (youtubeID:string) => {
+  let thumbnailLink = `https://img.youtube.com/vi/${youtubeID}/0.jpg`
+  return await fetch(thumbnailLink,{
+    method: 'GET',
+    redirect: 'follow'
+  })
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/utils/getSupportedFileTypeFromLink.ts:27]: have you tried it locally?

**Comment by UmairMukhtar** [comment]
@AltiMario i have tried with a link similar to your link and it worked for me please have a look

[Link Similar to Mario.webm](https://user-images.githubusercontent.com/34018917/222341897-d607435e-73cc-4e16-adc3-d050744d2ead.webm)

**Review by AltiMario — APPROVED** [review]
I cannot test properly because there is a backend component not published

File: app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx [hunk 2]
```diff
@@ -141,32 +136,6 @@ function NewMaterial({
...
-                  gap="8px"
-                  alignItems="center"
-                >
-                  <Box width="100%" className="bg-red color-white" padding="16px" borderRadius="12px" fontSize="16px">
-                    {linkError}
-                  </Box>
-                </Grid>
-              </>
-            )}
-
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx:169]: added again

File: app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx [hunk 3]
```diff
@@ -33,12 +33,7 @@ function NewMaterial({
   const user = authUser.getUser();
   const [editMode, setEditMode] = useState(false);
 
-  const { linkError, validateLink, isValidatingLink } = useLinkValidation({ customErrorMessage: 'Invalid material link' });
-
   const handleUpdate = async (values) => {
-    const response = await validateLink(values?.link);
-    if (!response) return;
-
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx:41]: added again

File: app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx [hunk 4]
```diff
@@ -200,7 +169,7 @@ function NewMaterial({
               }}
               type="submit"
             >
-              {isValidatingLink ? <Loader /> : 'Update File'}
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Creations/common/form/Steps/Two.jsx:203]: added again

File: app/web-frontend/src/pages/Creations/common/form/Steps/One.jsx [hunk 5]
```diff
@@ -14,14 +12,8 @@ export default function StepOne({
   onTagInputChange = () => {},
   tagSuggestions = [],
   status = {},
-  loading = false,
 }) {
-  const { linkError, validateLink, isValidatingLink } = useLinkValidation({ customErrorMessage: 'Invalid creation source link' });
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/web-frontend/src/pages/Creations/common/form/Steps/One.jsx:19]: added again

File: app/api/src/utils/getSupportedFileTypeFromLink.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import FileType from 'file-type';
 import httpStatus from 'http-status';
 import ApiError from './ApiError';
 import supportedMediaTypes from '../constants/supportedMediaTypes';
+var http = require('http');
```
[the remark this answers is not in the document]
Comment (UmairMukhtar) [reply app/api/src/utils/getSupportedFileTypeFromLink.ts]: fixed

**Comment by UmairMukhtar** [comment]
image

**Review by AltiMario — APPROVED** [review]

**Comment by AltiMario** [comment]
good point

**Comment by UmairMukhtar** [comment]
Creation Details Page
image

**Comment by UmairMukhtar** [comment]
Home Screen
image

**Comment by UmairMukhtar** [comment]
image

**Comment by UmairMukhtar** [comment]
image

**Comment by UmairMukhtar** [comment]
image

**Pull request merged**